mantidproject/mantid
# Improve usability of run loading on fitting tab of Engineering Diffraction UI

**Description of work.**

The File Finder widget on the Fitting tab has been updated to filter the files shown in the Browse dialog based on two combo boxes. The combo boxes allow the user to specify a unit (TOF or dSpacing) and\or a grouping (basically North or South bank):

![image](https://user-images.githubusercontent.com/56295817/126459236-d49a3690-5553-423c-8921-dbcf0662f013.png)

In order to get this to work a switch has been added to the FileFinderWidget that is fed through to the underlying QFileDialog object. QFileDialog can be set to use either a Qt file browse dialog or the OS-native file dialog. The Qt version is easier to customise and you can add custom file filters to it. On Windows at least it looks a bit more dated than the OS native dialog but hopefully overall this is an improvement. The switch is a method called FileFinderWidget::setUseNativeWidget

Note - there was already a unit dropdown prior to this PR. It used to drive a unit conversion into whatever unit was selected. This has been repurposed as a file filter and the unit conversion has been removed. The unit conversion feature is a hangover from an earlier version of the Focus tab where it used to output files in TOF only. The focussing now outputs TOF and dSpacing data so there's no longer a need for any unit conversion to occur

**To test:**

It's easiest to test if you have some focused data files already generated for a few runs. You'll need ISIS data archive access:

1. Go to Calibration tab
2. Fill in vanadium 236516 ceria 193749 
3. Hit Calibrate
4. Go to Focus tab
5. Enter 299080-299084
6. Hit Focus

Having done this go to the Fitting tab
Hit Browse
Based on an earlier PR (#32016) the folder should default to the folder where the focussing outputs the files. If for any reason this isn't what you see, go to the Engineering_Mantid\Focus folder in your user area (~ on linux or `C:\Users\<fedid>\ `on Windows)
You should see the set of focused .nxs files you just generated:

![image](https://user-images.githubusercontent.com/56295817/126462178-df307957-64d5-4d18-9026-bddced079c7f.png)

Observe that you only see a subset of file based on the choices made on the TOF or Grouping combo boxes

Finally, worth checking I've not broken the file finder widget by going to another user interface that uses eg Indirect, Corrections UI

Fixes #31690. 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
